src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowCreate.cs

```diff
@@ -112,16 +119,25 @@ public override DbCommand GetCommand(DbConnection connection)
                 {
                     continue;
                 }
-                
+
                 // Add the input column
                 inColumnNames.Add(metadata.EscapedName);
-                
+
                 // Add the input values as parameters
                 string paramName = $"@Value{RowId}_{i}";
-                inValues.Add(paramName);
-                inParameters.Add(new SqlParameter(paramName, column.SqlDbType) {Value = cell.Value});
+
+                if (metadata.IsHierarchyId)
```

Comment:
This part should definitely go into the utilities. I can't exactly remember how it all works right now. If I recall correctly, when we put together a query to add/update there's a utility (ToSqlScript) that turns the value into an expression SQL Server can work with. Take a look at the FormatDecimalLike method in that utility class to see an example of how we take decimals and use a `CAST` on them. There's even a line that calls out how hierarchyid doesn't work right now, so that might make it easier!